lib/actions/posts.ts

```diff
@@ -26,7 +59,42 @@ export async function getPosts(filterType: PostType, tag: string) {
   }
 }
 
-export async function getPost(id: number) {
-  const post = testPosts.find((post) => post.id === id);
-  return post;
+export async function getPost(id: string) {
+  await connectToDB();
+  const session = await getSession();
+  const sessionUser = session?.user;
```

Comment:
We don't seem to be using a session here.  If you want to make it so only the user who made the post can view it, then you can adjust this query to look for the creator as well. Otherwise let's remove this extra session info!

(Stretch goal: it would be cool to have a toggle for private/public posts that would let you share with other people )

Reply:
This is a cool idea that I want to get back to once the core functionality of the program is done.